nevergrad/functions/pownet/solver.py

```diff
@@ -0,0 +1,251 @@
+import os
+import importlib.util
+import requests
+from pyomo.opt import SolverFactory
+from pyomo.core import Var
+from operator import itemgetter
+from datetime import datetime
+import nevergrad as ng
+from nevergrad.functions.pyomo import core
+
+import git
+import pypownetr.model
+
+def download_dataset(dataset_name="cambodian", force_update=False):
+    #git clone <repository> <path>
+    dataset_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'pypownet')
+    if not os.path.exists(dataset_path):
+        git.Repo.clone_from("https://github.com/pacowong/pypownet.git", dataset_path, branch="main")
+    elif force_update:
+        git.cmd.Git(dataset_path).pull('origin', 'main') #Check update
+    if dataset_name == "cambodian":
+        return os.path.join(dataset_path, 'pypownetr', 'datasets', 'kamal0013', 'camb_2016')
+    else:
+        raise NotImplementedError
+
+def _load_sim_data(dst, src, nodes, horizon_hours, day):
+    for z in nodes:
+        for i in range(1, horizon_hours+1):
+            dst[z,i] = src[z,(day-1)*24+i]
+
+
+def solve_pownet_subproblems(pyomo_model, model_data_path, solver, year=2016, start_day=1, last_day=365):
+    """
+    simulation year, start(1-365) and end(1-365) days of simulation
+    """
+    
+    instance = pyomo_model.create_instance(model_data_path)
+
+    ###solver and number of threads to use for simulation
+    H = instance.HorizonHours
+    K = range(1, H+1)
+
+    ###Run simulation and save outputs
+    #Containers to store results
+    on = []
+    switch = []
+
+    mwh = []
+    hydro = []
+    solar = []
+    wind = []
+
+    hydro_import = []
+
+    srsv = []
+    nrsv = []
+
+    vlt_angle = []
+
+    system_cost = []
```

Comment:
this is a whole lot of variables :s sorry but I'll ask you to add type hints for them because knowing what they should be will be helpful for maintenance on the long run.